power_grid_model_c/power_grid_model/include/power_grid_model/component/load_gen.hpp

```diff
@@ -64,7 +64,14 @@ class LoadGen final : public std::conditional_t<is_gen, GenericGenerator, Generi
     using InputType = LoadGenInput<sym>;
     using UpdateType = LoadGenUpdate<sym>;
     using BaseClass = std::conditional_t<is_gen, GenericGenerator, GenericLoad>;
-    static constexpr char const* name = sym ? (is_gen ? "sym_gen" : "sym_load") : (is_gen ? "asym_gen" : "asym_load");
+    static constexpr char const* get_name() {
+        if constexpr (sym) {
+            return is_gen ? "sym_gen" : "sym_load";
+        } else {
+            return is_gen ? "asym_gen" : "asym_load";
+        }
+    }
```

Comment:
if you're not going to make this a lambda-function, please make this function private. no other user should look ever at this function and instead, the `name` constexpr variable should be used